bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/discovery/CaddxDiscoveryService.java

```diff
@@ -0,0 +1,218 @@
+import org.openhab.binding.caddx.internal.handler.CaddxBridgeHandler;
+import org.openhab.binding.caddx.internal.handler.CaddxThingType;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is responsible for discovering the supported Things.
+ *
+ * @author Georgios Moutsos - Initial contribution
+ */
+@Component(service = DiscoveryService.class, immediate = true, configurationPid = "discovery.caddx")
+@NonNullByDefault
+public class CaddxDiscoveryService extends AbstractDiscoveryService {
+    private final Logger logger = LoggerFactory.getLogger(CaddxDiscoveryService.class);
+
+    private @NonNullByDefault({}) SerialPortManager portManager;
+    private CaddxBridgeDiscovery caddxBridgeDiscovery = new CaddxBridgeDiscovery(portManager, this);
+    private @Nullable CaddxBridgeHandler caddxBridgeHandler;
+
+    /**
+     * Constructor.
+     *
+     * @param caddxBridgeHandler The Bridge handler
+     */
+    public CaddxDiscoveryService(CaddxBridgeHandler caddxBridgeHandler) {
+        super(CaddxBindingConstants.SUPPORTED_THING_TYPES_UIDS, 15, true);
+        this.caddxBridgeHandler = caddxBridgeHandler;
+    }
+
+    @Override
+    protected void startScan() {
+        logger.trace("Start Caddx Bridge discovery.");
+
+        if (portManager != null) {
+            caddxBridgeDiscovery = new CaddxBridgeDiscovery(portManager, this);
+            caddxBridgeDiscovery.discoverBridge();
+        }
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        super.stopScan();
+    }
+
+    /**
+     * Method to add a Caddx Bridge to the Smarthome Inbox.
+     *
+     * @param port
+     */
+    public void addCaddxBridge(CaddxProtocol protocol, String port, int baudrate) {
+        logger.trace("addCaddxBridge(): Adding new Caddx Bridge on {} {} to Smarthome inbox", port, baudrate);
+
+        String bridgeID = "";
+        boolean containsChar = port.contains("/");
+
+        if (containsChar) {
+            String[] parts = port.split("/");
+            String id = parts[parts.length - 1].toUpperCase();
+            bridgeID = id.replaceAll("\\W", "_");
+        } else {
+            String id = port.toUpperCase();
+            bridgeID = id.replaceAll("\\W", "_");
+        }
+
+        Map<String, Object> properties = new HashMap<>(0);
+        properties.put(CaddxBridgeConfiguration.PROTOCOL, protocol);
+        properties.put(CaddxBridgeConfiguration.SERIAL_PORT, port);
+        properties.put(CaddxBridgeConfiguration.BAUD, baudrate);
+
+        try {
+            ThingUID thingUID = new ThingUID(CaddxBindingConstants.CADDXBRIDGE_THING_TYPE, bridgeID);
+
+            thingDiscovered(DiscoveryResultBuilder.create(thingUID).withProperties(properties)
+                    .withLabel("Caddx Bridge - " + port).build());
+        } catch (Exception e) {
```

Comment:
what exception are you expecting here?

Reply:
This exception handler was copied from the corresponding class of the dscalarm binding. I removed it.